lib/src/yaml_edit/editor.dart

```diff
@@ -0,0 +1,651 @@
+      }
+    }
+
+    /// If [_indentation] was not set by the above block, give it a default
+    /// value of 2.
+    _indentation ??= 2;
+
+    return _indentation;
+  }
+
+  int _indentation;
+
+  String get lineEnding {
+    if (_isWindowsEnding == null) {
+      var index = -1;
+      var unixNewlines = 0;
+      var windowsNewlines = 0;
+      while ((index = _yaml.indexOf('\n', index + 1)) != -1) {
+        if (index != 0 && _yaml[index - 1] == '\r') {
+          windowsNewlines++;
+        } else {
+          unixNewlines++;
+        }
+      }
+      _isWindowsEnding = windowsNewlines > unixNewlines;
+    }
+
+    return _isWindowsEnding ? '\r\n' : '\n';
+  }
+
+  bool _isWindowsEnding;
+
+  factory YamlEditor(String yaml) => YamlEditor._(yaml);
+
+  YamlEditor._(this._yaml) {
+    ArgumentError.checkNotNull(_yaml);
+    initialize();
+  }
+
+  void initialize() {
+    _contents = loadYamlNode(_yaml);
+    _aliases = {};
+
+    var stack = [_contents];
+    var map = <YamlNode, bool>{};
+
+    while (stack.isNotEmpty) {
+      var nextElem = stack.removeLast();
+
+      if (map[nextElem] == null) {
+        map[nextElem] = true;
+      } else {
+        _aliases.add(nextElem);
+      }
+
+      if (nextElem is YamlMap) {
+        stack.addAll(nextElem.nodes.values);
+      } else if (nextElem is YamlList) {
+        stack.addAll(nextElem.nodes);
+      }
+    }
+
+    _isWindowsEnding = null;
+    _indentation = null;
+  }
+
+  /// Parses the document to return [YamlNode] currently present at [path]. If
+  /// no [YamlNode]s exist at [path], [parseAt] will return a [YamlNode]-wrapped
+  /// [orElse] if it is defined, or throw an [Error] otherwise. The value passed
+  /// to [orElse] has to be a valid YAML element (i.e. scalar/ list/ map).
+  ///
+  /// To get `null` when [path] does not point to a value in the [YamlNode]-tree,
+  /// simply pass `orElse: null`.
+  /// ```dart
+  /// final myYamlEditor('{"key": "value"}');
+  /// final value = myYamlEditor.valueAt(['invalid', 'path'], orElse: null);
+  /// print(value) // null
+  /// ```
+  ///
+  /// Common usage example:
+  /// ```dart
+  ///   final doc = YamlEditor('''
+  /// a: 1
+  /// b:
+  ///   d: 4
+  ///   e: [5, 6, 7]
+  /// c: 3
+  /// ''');
+  /// print(doc.parseAt(['b', 'e', 2])); // 7
+  /// ```
+  ///
+  /// To illustrate that [parseAt] returns a [YamlNode] that can be invalidated
+  /// if modifications are performed to the document,
+  /// ```dart
+  /// final doc = YamlEditor("YAML: YAML Ain't Markup Language");
+  /// final node = doc.parseAt(['YAML']);
+  ///
+  /// print(node.value); // Expected output: "YAML Ain't Markup Language"
+  ///
+  /// doc.assign(['YAML'], 'YAML');
+  ///
+  /// final newNode = doc.parseAt(['YAML']);
+  ///
+  /// // Note that the value does not change
+  /// print(newNode.value); // "YAML"
+  /// print(node.value); // "YAML Ain't Markup Language"
+  /// ```
+  YamlNode parseAt(Iterable<Object> path, {Object orElse = #noArg}) {
```

Comment:
Maybe make `orElse` a function computing the result. Also don't use a symbol, they are not the right thing to use here.

Like: https://api.dart.dev/stable/2.8.4/dart-async/Stream/firstWhere.html

Reply:
Modified `orElse` as per your suggestion!